presto-native-execution/presto_cpp/main/http/HttpClient.cpp

```diff
@@ -123,8 +125,12 @@ class ResponseHandler : public proxygen::HTTPTransactionHandler {
   ResponseHandler(
       const proxygen::HTTPMessage& request,
       velox::memory::MemoryPool* pool,
-      const std::string& body)
-      : request_(request), body_(body), pool_(pool) {
+      const std::string& body,
+      const std::function<void(int)> reportOnBodyStatsFunc)
```

Comment:
We can remove const qualifier, because you are moving it later. With const the copy constructor instead of move constructor will be called